### PR TITLE
browser(webkit): roll to 01/27/22

### PR DESCRIPTION
**Pull request by dpino**
Rebase `webkit/patches/bootstrap.diff` to [r288658](https://trac.webkit.org/changeset/288658/webkit).

  * 3-way diff: https://github.com/dpino/WebKit/commit/8395a315d5a29add03177b7ff7317f64a4cb9fe0
  * Resolve conflicts: https://github.com/dpino/WebKit/commit/32065ec8b4257e60b8bc52038ab75f3ddc82d908

### CODE REVIEW

File: browser_patches/webkit/embedder/Playwright/win/WebKitBrowserWindow.cpp [hunk 1]
```diff
@@ -385,20 +384,6 @@ void WebKitBrowserWindow::didNotHandleKeyEvent(WKPageRef, WKNativeEventPtr event
     PostMessage(thisWindow.m_hMainWnd, event->message, event->wParam, event->lParam);
 }
 
-void WebKitBrowserWindow::decidePolicyForNavigationAction(WKPageRef page, WKFrameRef frame, WKFrameNavigationType navigationType, WKEventModifiers modifiers, WKEventMouseButton mouseButton, WKFrameRef originatingFrame, WKURLRequestRef request, WKFramePolicyListenerRef listener, WKTypeRef userData, const void* clientInfo)
```
Comment (yury-s) [inline browser_patches/webkit/embedder/Playwright/win/WebKitBrowserWindow.cpp:388]: Please revert this, you remove recently added code.

File: browser_patches/webkit/patches/bootstrap.diff [hunk 2]
```diff
@@ -8746,19 +8746,19 @@ index 2e90534ffd8da83b7dc54d46fa7def16319bbb43..2493c00d58957751c65c37eb409fa8d6
...
-index 694008e0451edc5770142a0a6d9eed52b04ded80..ec93869f9486bdf7bd3bb56478c62469d2fa58b6 100644
+index 01bb6ef9e22e7c595588ab8b312792a08533aead..db8e2b16d00c91ea1d2fec0a41a7e8c32d571e75 100644
 --- a/Source/WebCore/rendering/ScrollAlignment.h
 +++ b/Source/WebCore/rendering/ScrollAlignment.h
-@@ -78,6 +78,7 @@ struct ScrollAlignment {
-     Behavior m_rectVisible;
+@@ -81,6 +81,7 @@ struct ScrollAlignment {
      Behavior m_rectHidden;
      Behavior m_rectPartial;
+     bool m_enableLegacyHorizontalVisibilityThreshold { true };
```
Comment (yury-s) [inline browser_patches/webkit/patches/bootstrap.diff]: Looks like m_enableLegacyHorizontalVisibilityThreshold and m_disableMinThreshold that we add in our patch are doing the same thing. We should try removing m_disableMinThreshold (in separate PR is fine).

Comment (dpino) [reply browser_patches/webkit/patches/bootstrap.diff]: Ok, removing `m_disableMinThreshold` required this extra change:

https://github.com/dpino/WebKit/commit/e42db6aa7c83298196cba2a585fb347ae771217c

File: browser_patches/webkit/embedder/Playwright/win/WebKitBrowserWindow.cpp (hunk 1 above)
Comment (dpino) [reply browser_patches/webkit/embedder/Playwright/win/WebKitBrowserWindow.cpp]: I was wondering why `export webkit` added these changes. I had to reapply the `bootstrap.diff` patch after rebase to top of `main`.

**Comment by dpino** [comment]
I updated the patch. Links to commits:

* 3-way diff: https://github.com/dpino/WebKit/commit/bb66fe95427c9d3cf044ca3a70a38463ae485081
* Resolve conflicts: https://github.com/dpino/WebKit/commit/d0995af66eed4cdb9ba09b64ec9c5ce932e02a31
* Build fix: https://github.com/dpino/WebKit/commit/e42db6aa7c83298196cba2a585fb347ae771217c

**Review by yury-s — APPROVED** [review]

File: browser_patches/webkit/patches/bootstrap.diff [hunk 3]
```diff
@@ -3522,7 +3522,7 @@ index 7a790bb23c8f37af57e4284b367cad9d679a434f..a8b090780ff0811a0ac7614ee6ad3da1
 +        absoluteBounds.setHeight(LayoutUnit(std::max(*height, 1.0)));
 +    }
 +    ScrollAlignment alignment = ScrollAlignment::alignCenterIfNeeded;
-+    alignment.m_disableMinThreshold = true; // Disable RenderLayer minium horizontal scroll threshold.
++    alignment.m_enableLegacyHorizontalVisibilityThreshold = true;
```
Comment (yury-s) [inline browser_patches/webkit/patches/bootstrap.diff]: Overlooked this, I assume the value assigned to the new variable should be `false`

Comment (yury-s) [reply browser_patches/webkit/patches/bootstrap.diff]: Fixed in https://github.com/microsoft/playwright/pull/11722

**Pull request merged**